Client/Frontend/ContentBlocker/ContentBlockerHelper.swift

```diff
@@ -88,62 +92,59 @@ class ContentBlockerHelper {
     static private var blockImagesRule: WKContentRuleList?
     static var heavyInitHasRunOnce = false
 
-    // Only set and used in UI test
-    static weak var testInstance: ContentBlockerHelper?
-
     init(tab: Tab, profile: Profile) {
         self.tab = tab
         self.userPrefs = profile.prefs
 
-        if AppConstants.IsRunningTest {
-            ContentBlockerHelper.testInstance = self
-        }
+        NotificationCenter.default.addObserver(self, selector: #selector(setupTabTrackingProtection), name: .ContentBlockerTabSetupRequired, object: nil)
 
-        if ContentBlockerHelper.heavyInitHasRunOnce {
+        guard let prefs = userPrefs, !ContentBlockerHelper.heavyInitHasRunOnce else {
             return
         }
 
+        performHeavyOneTimeInit(prefs)
+    }
+
+    private func performHeavyOneTimeInit(_ prefs: Prefs) {
+        struct RunOnce { static var hasRun = false }
+        if RunOnce.hasRun { return }
+        RunOnce.hasRun = true
```

Comment:
Re-entrancy guard is needed, I should have caught this earlier. Alternative is to wrap the init code up into a class of its own, but seems like overkill. This is my boilerplate re-entrancy guard syntax, not sure if we have a specific style I should follow.

Reply:
Yeah, half the code has static scope and the other half per-tab, and could be re-organized around that, and now with TP moving from a rarely-used feature to highly-used, it has a role in the normal launch flow.
I think this should be a follow up bug, I have other feature branches based off of this code I would like to finish before getting to that.